src/java/org/apache/cassandra/index/sasi/analyzer/DelimiterAnalyzer.java

```diff
@@ -56,14 +57,16 @@ public ByteBuffer next()
         return iter.next();
     }
 
+    public void validate(Map<String, String> options, AbstractType validator) throws ConfigurationException
+    {
+        if (!VALID_ANALYZABLE_TYPES.containsKey(validator))
+            throw new ConfigurationException(String.format("Only text types supported, got %s", validator));
```

Comment:
I think this error is unreachable. The new `AbstractAnalyzer#validate` method is called exclusively from `IndexMode#validateAnalyzer`, which already validates the type with a call to `AbstractAnalyzer#isCompatibleWith`. 

This makes me thing that probably we don't need the cell type argument on `AbstractAnalyzer#validate`, or alternatively we could consider consider making `AbstractAnalyzer#validate` consistently responsible for checking the cell type so `IndexMode#validateAnalyzer` doesn't have to call `AbstractAnalyzer#isCompatibleWith`. wdyt?

Reply:
@adelapena 
I do not have a problem with the approach you are describing at the end of your response but please look closely here:

````
                if (!analyzer.isCompatibleWith(cd.type))
                    throw new ConfigurationException(String.format("%s does not support type %s",
                                                                   analyzerClass.getSimpleName(),
                                                                   cd.type.asCQL3Type()));
                analyzer.validate(indexOptions, cd.cellValueType());
````

`cd.type` is not the same thing as `cd.cellValueType()` (go into cellValueType() and read the docs).

isCompatibleWith excepts 'different" AbstractType than validate method would accept, if I understand that correctly.

Reply:
I missed that one. However, I still don't see a use case where that second configuration exception could be raised, since SASI doesn't support non-frozen collections or UDTs. If the exception where reachable I wonder if we would need a similar check in other text analyzer, and that would make the case to pass the cell value type, or the entire column definition, to `AbstractAnalyzer#isCompatibleWith`.
I might be missing something obvious, but I think that this check is a remanent from what there was before creating the `isCompatibeWith` method during CASSANDRA-13669. wdyt?